waifu2x.ts

```diff
@@ -827,4 +877,53 @@ export default class Waifu2x {
     }
 }
 
+export type AsyncCallback<T> = () => Promise<T>;
+export class AsyncSemaphore {

```

Comment:
Coding style:
-Do not use semicolons.
-Explicitly label public methods as `public`.
-Please use arrow function syntax for all methods.